crypto/x509/x509_test.cc

```diff
@@ -201,6 +201,31 @@ Z0IL+OQFz6+LcTHxD27JJCebrATXZA0wThGTQDm7crL+a+SujBY=
 -----END CERTIFICATE-----
 )";
 
+// kExampleRsassaPssCert is an example RSA-PSS self-signed certificate,
+// signed with sha256. The Public Key Algorithm of 'kExamplePSSCert' is 'rsaEncryption'.
+// But the Public Key Algorithm of'kExampleRsassaPssCert' is 'rsassaPss'.
+static const char kExampleRsassaPssCert[] = R"(
+-----BEGIN CERTIFICATE-----
```

Comment:
Hmm, RFC specifies 4 elements in the parameter sequence: https://tools.ietf.org/html/rfc8017#appendix-A.2.3

But when I decode I only see 3 elements:
https://lapo.it/asn1js/#MIIDDDCCAcMCFHg_A8W4Ltk2dAz6XmuAzRhcy2AzMD4GCSqGSIb3DQEBCjAxoA0wCwYJYIZIAWUDBAIBoRowGAYJKoZIhvcNAQEIMAsGCWCGSAFlAwQCAaIEAgIA3jANMQswCQYDVQQDDAJDQTAgFw0xOTEyMTgxOTIyMjZaGA8yMTE5MTEyNDE5MjIyNlowFjEUMBIGA1UEAwwLZXhhbXBsZS5jb20wggEgMAsGCSqGSIb3DQEBCgOCAQ8AMIIBCgKCAQEArRRhhPBrpx0Wyu_RPzG1krirpQO8FScrFL3efZzc0wPJhmiynHHooIiPemrumxKTIFEJn7xH_uVNYPl0ae2UvvZ9wF2Avnsmgj2i9sEMUf_T2mba1hyDPukpGyTdwyBNXHoOsVG2-WrhMW0NkGxOJwdvqK7k_fmNRW54p5cdf-oXBICd3ZeXUGwN-_O9P8AUR-swPF5_xQT0l6cIL8p1Ii3IOwQbsbIclwcgB-A-JxMBOrsS2t3wmO_t00nYJQQ39GvFOG7FLMt771bG_lkXfE-uM9BPqORk63XJP5GYtI0G9yc8KjJffjMnELfyNu9y6h-_jheUxnol-M4ZlVyBNQIDAQABMD4GCSqGSIb3DQEBCjAxoA0wCwYJYIZIAWUDBAIBoRowGAYJKoZIhvcNAQEIMAsGCWCGSAFlAwQCAaIEAgIA3gOCAQEACYrASuTnV7G0Fs4YD4oYTHPtP3hmXnOqLYFwg5mkD1jNUQAoZ1228iswPduZQgO97A2vhI_qFW1efyWdElqC9RrTzldtdS9nAMeOJ6m-dXNfHbsPjZl2-g-kreAxCO7_-tdZ_TBqAS6llEIo7EqMGPeE-HwwHa2Jv5La7HAT_KNA3jCGaucTGVV77CgWVc_cNJq2EIf9DOe2-zokWnMQxbV84ks26uQy3CfF8UZrxwcr3aKJxzR29u2Io6vvnTs7dJZFafFLwtdLF6n7_OuycizeY0z_PjBnZN7Xw3pTDruglf9wNco1m_hMdLqJ8XTi_6wGTUyETfm7J2j9U1yy9g

Reply:
This test input is from s2n. The spec rfc8017 does not require all fields MUST exist.

```
trailerField is the trailer field number, for compatibility with
      IEEE 1363a [IEEE1363A].  It SHALL be 1 for this version of the
      document, which represents the trailer field with hexadecimal
      value 0xbc.
```

Reply:
That's right, thanks. But when you add pss param parsing, you should definitely add test cases that runs through all valid combinations of parameters.

Reply:
Will do.

Reply:
I added diff params in PR#89 except trailer field, which is required to be omitted during generation
https://tools.ietf.org/html/rfc4055#page-7
```
Implementations that perform signature generation MUST omit the trailerField field, indicating that the default trailer field value was used.
```